src/librbd/WatchNotifyTypes.h

```diff
@@ -219,24 +219,23 @@ struct SnapPayloadBase {
   static const bool CHECK_FOR_REFRESH = true;
 
   std::string snap_name;
+  cls::rbd::SnapshotNamespace snap_namespace;
 
   void encode(bufferlist &bl) const;
   void decode(__u8 version, bufferlist::iterator &iter);
   void dump(Formatter *f) const;
 
 protected:
   SnapPayloadBase() {}
-  SnapPayloadBase(const std::string &name) : snap_name(name) {}
+  SnapPayloadBase(const std::string &name, const cls::rbd::SnapshotNamespace& _snap_namespace) : snap_name(name), snap_namespace(_snap_namespace) {}
```

Comment:
Nit: avoid going past 80 characters in a single line. This comment applies to all changes in this file.